common/src/main/java/com/wynntils/wynn/model/container/ScriptedContainerQuery.java

```diff
@@ -0,0 +1,203 @@
+            query.setErrorHandler(errorHandler);
+            return this;
+        }
+
+        public QueryBuilder expectTitle(String expectedTitle) {
+            if (verification != null) {
+                throw new IllegalStateException("Set verification twice");
+            }
+            this.verification = (title, type) -> title.getString().equals(expectedTitle);
+            checkForCompletion();
+            return this;
+        }
+
+        public QueryBuilder matchTitle(String regExp) {
+            if (verification != null) {
+                throw new IllegalStateException("Set verification twice");
+            }
+            this.verification = (title, type) -> title.getString().matches(regExp);
+            checkForCompletion();
+            return this;
+        }
+
+        public QueryBuilder processContainer(ContainerAction action) {
+            if (handleContent != null) {
+                throw new IllegalStateException("Set handleContent twice");
+            }
+            this.handleContent = action;
+            checkForCompletion();
+            return this;
+        }
+
+        public QueryBuilder clickOnSlot(int slotNum) {
+            if (startAction != null) {
+                throw new IllegalStateException("Set startAction twice");
+            }
+            this.startAction = (container) -> {
+                ContainerUtils.clickOnSlot(slotNum, container.containerId(), container.items());
+                return true;
+            };
+            checkForCompletion();
+            return this;
+        }
+
+        public QueryBuilder clickOnSlotMatching(int slotNum, Item expectedItemType, String expectedItemName) {
+            if (startAction != null) {
+                throw new IllegalStateException("Set startAction twice");
+            }
+            this.startAction = (container) -> {
+                ItemStack item = container.items().get(slotNum);
+                if (!item.is(expectedItemType)
+                        || !item.getDisplayName().getString().equals(expectedItemName)) return false;
+
+                ContainerUtils.clickOnSlot(slotNum, container.containerId(), container.items());
+                return true;
+            };
+            checkForCompletion();
+            return this;
+        }
+
+        public QueryBuilder useItemInHotbar(int slotNum) {
+            if (startAction != null) {
+                throw new IllegalStateException("Set startAction twice");
+            }
+            this.startAction = (container) -> ContainerUtils.openInventory(slotNum);
+            checkForCompletion();
+            return this;
+        }
+
+        public ScriptedContainerQuery build() {
+            if (startAction != null || handleContent != null || verification != null) {
+                throw new IllegalStateException("Partial contents only for last step");
+            }
+            return query;
+        }
+
+        private void checkForCompletion() {
```

Comment:
The fact that creating multiple steps in the builder is done implicitly should be commented

Reply:
I added javadoc to QueryBuilder. I hope you find it acceptable.